www.easyjet.com - Add Quirks to see flight map

karlcow opened the pull request:
#### 6995ccc4ca454b47c78e84ac2729ab7043545c99
<pre>
www.easyjet.com - Add Quirks to see flight map
<a href="https://bugs.webkit.org/show_bug.cgi?id=274374">https://bugs.webkit.org/show_bug.cgi?id=274374</a>
<a href="https://rdar.apple.com/126452111">rdar://126452111</a>

Reviewed by Brent Fulgham.

EasyJet website has a map system which is dependent on Bing.
This call to bing doesn&apos;t work when the user agent is Safari, but is
working correctly what the user agent is switched to Firefox or Chrome.
So all requests to easyjet double check if there is a secondary call to
bing and overrides it with a Firefox User Agent.

Note: That the solution would be more elegant if there was a unified
system for user agent override, hence the FIXME.

* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::userAgent const):

Canonical link: <a href="https://commits.webkit.org/292922@main">https://commits.webkit.org/292922@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5db7d0adf726fe31ce39e9fa96358500a507f469

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/97474 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/17099 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/7315 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/102561 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/48003 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/99519 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/17392 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/25551 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/74274 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31454 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/100477 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13168 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/88149 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54619 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/12936 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/6031 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/47445 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/82967 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/6112 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/104581 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/24553 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/17908 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/83318 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/24925 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/84275 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82739 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20815 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27281 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/4969 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/18135 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/24515 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/29684 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/24337 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/27651 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/25911 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

karlcow commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -3187,7 +3187,17 @@ String FrameLoader::userAgent(const URL& url) const
+    // FIXME: Move this to a proper UA override singular mechanism
+    // https://bugs.webkit.org/show_bug.cgi?id=274374
+    if (RefPtr document = m_frame->document()) {
+        if (RegistrableDomain(document->topURL()).string() == "easyjet.com"_s) {
```
we can probably constraint this a bit more by matching on the path containing `routemap`.

karlcow replied:
this is fixed.

brentfulgham commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -3187,7 +3187,17 @@ String FrameLoader::userAgent(const URL& url) const
-    
+
+    // FIXME: Move this to a proper UA override singular mechanism
+    // https://bugs.webkit.org/show_bug.cgi?id=274374
```
I'd move this comment down to the Firefox string below, and use this location for our normal FIXME about removing the Quirk once we get Easyjet (or is it Bing?) to fix their site.

brentfulgham reviewed (approved):
These Quirks are always gross, but this certainly fixes the site. r=me

karlcow committed:
www.easyjet.com - Add Quirks to see flight map
https://bugs.webkit.org/show_bug.cgi?id=274374
rdar://126452111

Reviewed by Brent Fulgham.

EasyJet website has a map system which is dependent on Bing.
This call to bing doesn't work when the user agent is Safari, but is
working correctly what the user agent is switched to Firefox or Chrome.
So all requests to easyjet double check if there is a secondary call to
bing and overrides it with a Firefox User Agent.

Note: That the solution would be more elegant if there was a unified
system for user agent override, hence the FIXME.

* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::userAgent const):

Canonical link: https://commits.webkit.org/292922@main
